core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3401,11 +3409,39 @@ public PhysicalOperation visitTableExecute(TableExecuteNode node, LocalExecution
             return new PhysicalOperation(operatorFactory, outputMapping.buildOrThrow(), context, source);
         }
 
-        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, PlanNode source)
+        private int getWriterCount(Session session, Optional<PartitioningScheme> partitioningScheme, Optional<PartitioningScheme> preferredPartitioningScheme, PlanNode source)
         {
+            // This check is required because we don't know which writer count to use when exchange is
```

Comment:
nit: we could extract decision logic `part scaled OR unpart scaled OR part OR unpart` from `AddLocalExchanges` and not rely o plan node serach